compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -103,38 +107,95 @@ chooseDataTiledMMAAttr(TypeRange eTypes, IREE::GPU::TargetAttr target,
   // and each intrinsic consumes only 32 bits from A and B, then we want to set
   // unrollK=4 to turn 4 separate 32-bit loads into one 128-bit load.
   const int unrollK =
-      kLoadInstructionBits /
-      std::min(
-          intrinsicA.getElementTypeBitWidth() * intrinsicA.getNumElements(),
-          intrinsicB.getElementTypeBitWidth() * intrinsicB.getNumElements());
+      std::max(1, *wgp.getMaxLoadInstructionBits() /
+                      std::min(sizeInBits(intrinsicA), sizeInBits(intrinsicB)));
```

Comment:
That is the case in practice so far, and if it were always the case, the `std::max(1, ...)` would not be needed!

I added a testcase where that is not the case, and in that case, we would want the current behavior. The value 1 in that case simply says no unrolling on K in that case.

Since that is shorter than even writing that assertion, I don't feel a need to change that.

Reply:
done. thanks!